fvm/meter/weighted_meter.go

```diff
@@ -417,3 +446,95 @@ func (m *WeightedMeter) MemoryIntensities() MeteredMemoryIntensities {
 func (m *WeightedMeter) TotalMemoryEstimate() uint64 {
 	return m.memoryEstimate
 }
+
+// MeterStorageRead captures storage read bytes count and returns an error
+// if it goes beyond the total interaction limit and limit is enforced
+func (m *WeightedMeter) MeterStorageRead(storageKey StorageInteractionKey,
+	value flow.RegisterValue,
+	enforceLimit bool) error {
+
+	// all reads are on a View which only read from storage at the first read of a given key
+	if _, ok := m.storageUpdateSizeMap[storageKey]; !ok {
+		m.storageReadCounter++
+		readByteSize := getStorageKeyValueSize(storageKey, value)
+		m.totalStorageBytesRead += readByteSize
+		m.storageUpdateSizeMap[storageKey] = readByteSize
+	}
+
+	if enforceLimit &&
+		m.TotalBytesOfStorageInteractions() > m.storageInteractionLimit {
+		return errors.NewStorageCapacityExceededError(
```

Comment:
the error return in state is different? errors.NewLedgerInteractionLimitExceededError

also maybe refactor the enforce limit check into method

Reply:
addressed.